project-koku/koku
# Adding Azure to Sources Client integration

Adds Azure to the Koku-Sources Client integration service.  Platform-Sources stores `client_id`, `client_secret`, and `tenant_id`.  In Koku's current provider on boarding we require these values in addition to `subscription_id`, `storage_group`, and `storage_account`.

We must supply the `storage_group` and `storage_account` values to the existing `/billing_source` Sources Client HTTP endpoint.  A new endpoint `/authentications` was added to supply the `subscription_id`.

**Testing**
1. Create Azure Source in Platform-Sources backend
2.  Associate Storage Information with Azure Source.
3.  Associate Subscription ID with Azure Source.  Verify provider is created
4.  Add Azure Provider to Koku directly, then add source.  Verify that koku_uuid is updated in source.


**Test Results**
[azure_sources_ut.txt](https://github.com/project-koku/koku/files/3636818/azure_sources_ut.txt)
